In [1]:
import json
import numpy as np
import pandas as pd
import psycopg2

%load_ext jupyternotify

<IPython.core.display.Javascript object>

# How many doctors on Medicaid drug utilization boards have received payments from drug manufacturers?

We will use Open Payments data to determine the extent of payments from August 2013 through 2017 from drug manufacturers to doctors serving on Medicaid drug utilization boards between the start of 2016 and March of 2018.

## Create tables for analysis

First, connect to the database.

In [2]:
with open("config.json") as f:
    conf = json.load(f)

In [3]:
conn_str = "host={} dbname={} user={} password={}".format(conf["host"], conf["database"], conf["user"], conf["password"])

In [4]:
conn = psycopg2.connect(conn_str)
conn.autocommit = True # Allow the notebook to commit transactions (like creating a table) to the connected database.

Query the database to create a new table of all Medicaid drug utilization board members and the general purpose payments they received from drug manufacturers.

In [5]:
%%notify
payments_by_doctor = pd.read_sql("""CREATE TABLE IF NOT EXISTS payments_by_doctor AS
SELECT physician_profile_id1 AS physician_profile_id,
       first_name,
       last_name,
       state_name,
       degree_1,
       degree_2,
       payments_amount,
       number_of_payments
FROM
  (SELECT physician_profile_id AS physician_profile_id1,
          first_name,
          last_name,
          state_name,
          degree_1,
          degree_2
   FROM committee_members
   LEFT JOIN open_payments_general ON first_name = physician_first_name
   AND last_name = physician_last_name
   AND state_name = recipient_state
   GROUP BY physician_profile_id,
            first_name,
            last_name,
            state_name,
            degree_1,
            degree_2) AS matching_committee_members
LEFT JOIN
  (SELECT physician_profile_id AS physician_profile_id2,
          sum(total_amount_of_payment_usdollars) AS payments_amount,
          count(DISTINCT record_id) AS number_of_payments
   FROM open_payments_general
   GROUP BY physician_profile_id) AS totals_by_id ON matching_committee_members.physician_profile_id1 = totals_by_id.physician_profile_id2;

GRANT ALL ON TABLE payments_by_doctor TO redash;


SELECT *
FROM payments_by_doctor;""", con=conn)

<IPython.core.display.Javascript object>

The only consistent data we have about members serving on state drug utilization boards are the members' first name, last name and state. As such, that’s all we have to join against the Open Payments data. So in order to understand the extent of possible duplicates in this data, we will count how many distinct doctors in both the Open Payments general payments table and the [NPPES](https://npiregistry.cms.hhs.gov/) (a national dataset of physicians) registry table match our committee members.

In [6]:
%%notify
payments_by_doctor_counts = pd.read_sql("""
CREATE TABLE IF NOT EXISTS payments_by_doctor_counts AS
SELECT payments_by_doctor.physician_profile_id,
       payments_by_doctor.first_name,
       payments_by_doctor.last_name,
       payments_by_doctor.state_name,
       payments_by_doctor.degree_1,
       payments_by_doctor.degree_2,
       payments_amount,
       number_of_payments,
       doc_counts_nppes,
       doc_counts_open_payments
FROM payments_by_doctor
LEFT JOIN
  (SELECT provider_first_name,
          provider_last_name_legal_name,
          provider_business_practice_location_address_state_name,
          count(DISTINCT npi) AS doc_counts_nppes
   FROM nppes
   GROUP BY provider_first_name,
            provider_last_name_legal_name,
            provider_business_practice_location_address_state_name) AS nppes_grouped_by_first_last_state ON payments_by_doctor.first_name = nppes_grouped_by_first_last_state.provider_first_name
AND payments_by_doctor.last_name = nppes_grouped_by_first_last_state.provider_last_name_legal_name
AND payments_by_doctor.state_name = nppes_grouped_by_first_last_state.provider_business_practice_location_address_state_name
LEFT JOIN
  (SELECT physician_first_name,
          physician_last_name,
          recipient_state,
          count(DISTINCT physician_profile_id) AS doc_counts_open_payments
   FROM open_payments_general
   GROUP BY physician_first_name,
            physician_last_name,
            recipient_state) AS open_payments_grouped_by_first_last_state ON payments_by_doctor.first_name = open_payments_grouped_by_first_last_state.physician_first_name
AND payments_by_doctor.last_name = open_payments_grouped_by_first_last_state.physician_last_name
AND payments_by_doctor.state_name = open_payments_grouped_by_first_last_state.recipient_state;


GRANT ALL ON TABLE payments_by_doctor_counts TO redash;


SELECT *
FROM payments_by_doctor_counts;""", con=conn)

<IPython.core.display.Javascript object>

OK. So, we know we have some committee member doctors with the same name as other doctors in their state. Now let's find out the range of payments those doctors may have taken so we can get a sense as to how much uncertainty there is in our data vis-a-vis the prevalence of committee member doctors taking drug company money.

In [7]:
%%notify
payments_by_doctor_max_min = pd.read_sql("""
CREATE TABLE IF NOT EXISTS payments_by_doctor_max_min AS
SELECT physician_profile_id,
       first_name,
       last_name,
       state_name,
       degree_1,
       degree_2,
       payments_amount,
       max_amount,
       min_amount,
       number_of_payments,
       doc_counts_nppes,
       doc_counts_open_payments
FROM payments_by_doctor_counts
LEFT JOIN
  (SELECT first_name AS first_name_1,
          last_name AS last_name_1,
          state_name AS state_name_1,
          max(payments_amount) AS max_amount,
          min(payments_amount) AS min_amount
   FROM payments_by_doctor
   GROUP BY first_name_1,
            last_name_1,
            state_name_1) AS open_payments_grouped_by_first_last_state ON first_name = open_payments_grouped_by_first_last_state.first_name_1
AND last_name = open_payments_grouped_by_first_last_state.last_name_1
AND state_name = open_payments_grouped_by_first_last_state.state_name_1;

GRANT ALL ON TABLE payments_by_doctor_max_min TO redash;


SELECT *
FROM payments_by_doctor_max_min;""", con=conn)

<IPython.core.display.Javascript object>

When we joined the committee members table to the Open Payments table, our query returned multiple doctors with the same name and state in instances where multiple doctors were recorded as accepting payments. In order to avoid overstating the amount of payments these committee members took, we need to return a table of board members and the payments they received that eliminates duplicates.

In [8]:
%%notify
deduplicated_payments_by_doctor = pd.read_sql("""
CREATE TABLE IF NOT EXISTS deduplicated_payments_by_doctor AS
SELECT *
FROM
  (SELECT DISTINCT ON (first_name,
                       last_name,
                       state_name) *
   FROM payments_by_doctor_max_min) AS grouped_by_first_last_state;

GRANT ALL ON TABLE deduplicated_payments_by_doctor TO redash;


SELECT *
FROM deduplicated_payments_by_doctor;""", con=conn)

<IPython.core.display.Javascript object>

And finally, let's play it safe and remove those board member doctors who might have another doctor in the same state with the same name. To do so, we'll return a table that only includes doctors a) without possible duplicates, b) with more records in the Open Payments table than in the NPPES table or c) appearing only in the Open Payments table and not in the NPPES table.

In [9]:
%%notify
duplicates_removed = pd.read_sql("""
CREATE TABLE IF NOT EXISTS duplicates_removed AS
SELECT physician_profile_id, first_name, last_name, state_name, degree_1, degree_2, min_amount AS payments_amount, number_of_payments
FROM deduplicated_payments_by_doctor
WHERE doc_counts_nppes = doc_counts_open_payments
  AND doc_counts_nppes = 1
  OR doc_counts_nppes < doc_counts_open_payments
  OR doc_counts_nppes IS NULL
  AND doc_counts_open_payments = 1;

GRANT ALL ON TABLE duplicates_removed TO redash;


SELECT *
FROM duplicates_removed;""", con=conn)

<IPython.core.display.Javascript object>

## Calculate the proportion of doctors accepting drug company money

How much money from drug companies have doctors sitting on drug utilization boards received? We will use the minimum amount column so as to avoid overcounting payments in cases where more than one doctor with the same name in the same state is listed in Open Payments.

In [10]:
duplicates_removed["payments_amount"].sum()

3036013.1100000013

OK. So doctors sitting on drug utilization boards have taken in about $3 million. And how many of these doctors are there?

In [11]:
duplicates_removed[duplicates_removed["payments_amount"] > 0].count()

physician_profile_id    283
first_name              283
last_name               283
state_name              283
degree_1                274
degree_2                 29
payments_amount         283
number_of_payments      283
dtype: int64

OK. So, 283 doctors on drug utilization boards have taken at least some money from drug companies. What amount of all doctors sitting on these boards does that figure represent? To figure that out, we'll need to first calculate all the covered physicians sitting on the boards, as determined by their degree(s).

In [12]:
covered_degrees = ["MD", "DO", "DMD", "DDS", "DPM", "OD", "DC"]
covered_degrees

['MD', 'DO', 'DMD', 'DDS', 'DPM', 'OD', 'DC']

So how many doctors sitting on these boards either held one of these degrees or took money from drug companies (and were thus by definition covered by Open Payments)? That's our denominator.

In [13]:
deduplicated_payments_by_doctor[(deduplicated_payments_by_doctor["degree_1"].isin(covered_degrees)) | (deduplicated_payments_by_doctor["degree_2"].isin(covered_degrees)) | (deduplicated_payments_by_doctor["min_amount"] > 0)].count()

physician_profile_id        313
first_name                  455
last_name                   455
state_name                  455
degree_1                    445
degree_2                     44
payments_amount             313
max_amount                  313
min_amount                  313
number_of_payments          313
doc_counts_nppes            377
doc_counts_open_payments    313
dtype: int64

OK. So, it looks like 455 doctors either held one of these covered degrees or took money from drug companies (and were thus by definition covered by Open Payments). It's possible some of the doctors for whom we lack degree information are covered by Open Payments but we just don't know. (It's also possible that some of the doctors we excluded from our count of doctors taking drug company money because there was at least one possible duplicate who didn't take money did, in fact, take money.) In any case, using the 455 figure as our denominator, 62 percent of covered doctors sitting on these boards took some amount of money from drug companies.

How many received more than $1,000?

In [14]:
duplicates_removed[duplicates_removed["payments_amount"] > 1000].count()

physician_profile_id    103
first_name              103
last_name               103
state_name              103
degree_1                 99
degree_2                  9
payments_amount         103
number_of_payments      103
dtype: int64

OK. So, 103 doctors received more than $1,000.

How many received more than $10,000?

In [15]:
duplicates_removed[duplicates_removed["payments_amount"] > 10000].count()

physician_profile_id    34
first_name              34
last_name               34
state_name              34
degree_1                31
degree_2                 3
payments_amount         34
number_of_payments      34
dtype: int64

OK. So, 34 doctors received more than $10,000.

## Calculate the proportion of states on whose committees sit doctors accepting drug company money

Now let's break a couple of these figures down by state. First up: how many states had doctors on such a board who received money from drug companies?

In [16]:
# Calculate the number of doctors receiving money per state and the amount they took.
receiving_docs_by_state = duplicates_removed.groupby("state_name").agg({"payments_amount": [sum, max], "physician_profile_id": len})
receiving_docs_by_state.columns = receiving_docs_by_state.columns.droplevel() # Drop the first level of the multi-level column names
receiving_docs_by_state.reset_index(inplace=True)
receiving_docs_by_state.rename(columns={"sum": "total_payments", "max": "max_payment", "len": "doctors_paid"}, inplace=True)

# Calculate the total number of doctors per state.
total_docs_by_state = deduplicated_payments_by_doctor[(deduplicated_payments_by_doctor["degree_1"].isin(covered_degrees)) | (deduplicated_payments_by_doctor["degree_2"].isin(covered_degrees)) | (deduplicated_payments_by_doctor["min_amount"] > 0)].groupby("state_name").agg({"physician_profile_id": len})
total_docs_by_state.reset_index(inplace=True)
total_docs_by_state.rename(columns={"physician_profile_id": "total_doctors"}, inplace=True)

# Merge the dataframes.
payments_by_state = total_docs_by_state.merge(receiving_docs_by_state, how="outer", on="state_name")
payments_by_state[payments_by_state["total_payments"] > 0].count()

state_name        48
total_doctors     48
total_payments    48
max_payment       48
doctors_paid      48
dtype: int64

OK. So it looks like 48 states had doctors on such boards who received money.

And how many states had at least one doctor on such a board who received more than $1,000?

In [17]:
payments_by_state[payments_by_state["max_payment"] > 1000].count()

state_name        38
total_doctors     38
total_payments    38
max_payment       38
doctors_paid      38
dtype: int64

OK. It looks like 38 states had at least one doctor on such a board who received more than $1,000.

## Analysis of specific doctors

There are a handful of doctors whose payments particularly interest us.

First, how much Otsuka money did Dr. Divyansu Patel accept between January 2016 and the meeting of the Texas Drug Utilization Review Board on November 3, 2017 at which he spoke favorably of their drug Rexulti?

In [18]:
%%notify
patel_payments = pd.read_sql("""
CREATE TABLE IF NOT EXISTS patel_payments AS
SELECT date_of_payment,
       applicable_manufacturer_or_applicable_gpo_making_payment_name,
       name_of_drug_or_biological_or_device_or_medical_supply_1,
       nature_of_payment_or_transfer_of_value,
       sum(total_amount_of_payment_usdollars) as payments_amount
FROM open_payments_general
WHERE physician_last_name = 'PATEL'
  AND physician_first_name = 'DIVYANSU'
  AND recipient_state = 'TX'
  AND applicable_manufacturer_or_applicable_gpo_making_payment_name ILIKE '%OTSUKA%'
  AND date_of_payment::date >= '01/01/2016'
  AND date_of_payment::date <= '11/03/2017'
GROUP BY date_of_payment,
         applicable_manufacturer_or_applicable_gpo_making_payment_name,
         name_of_drug_or_biological_or_device_or_medical_supply_1,
         nature_of_payment_or_transfer_of_value;

GRANT ALL ON TABLE patel_payments TO redash;


SELECT *
FROM patel_payments;""", con=conn)

<IPython.core.display.Javascript object>

In [19]:
patel_payments["payments_amount"].sum()

12994.249999999998

OK. It looks like Dr. Patel accepted nearly $13,000 from Otsuka between January 2016 and November 3, 2017.

Next, how much money did Dr. Matthew Brams, another board member praising such injectable anti-psychotic medications, accept in that same time frame from drug makers Alkermes and Otsuka to speak about their products?

In [20]:
%%notify
brams_payments = pd.read_sql("""
CREATE TABLE IF NOT EXISTS brams_payments AS
SELECT date_of_payment,
       applicable_manufacturer_or_applicable_gpo_making_payment_name,
       name_of_drug_or_biological_or_device_or_medical_supply_1,
       nature_of_payment_or_transfer_of_value,
       sum(total_amount_of_payment_usdollars) AS payments_amount
FROM open_payments_general
WHERE physician_last_name = 'BRAMS'
  AND physician_first_name = 'MATTHEW'
  AND recipient_state = 'TX'
  AND (applicable_manufacturer_or_applicable_gpo_making_payment_name ILIKE '%OTSUKA%'
       OR applicable_manufacturer_or_applicable_gpo_making_payment_name ILIKE '%ALKERMES%')
  AND (name_of_drug_or_biological_or_device_or_medical_supply_1 ILIKE '%INVEGA TRINZA%'
       OR name_of_drug_or_biological_or_device_or_medical_supply_1 ILIKE '%ARISTADA%'
       OR name_of_drug_or_biological_or_device_or_medical_supply_1 ILIKE '%ABILIFY MAINTENA%')
  AND (nature_of_payment_or_transfer_of_value = 'Compensation for services other than consulting, including serving as faculty or as a speaker at a venue other than a continuing education program'
       OR nature_of_payment_or_transfer_of_value = 'Honoraria')
  AND date_of_payment::date >= '01/01/2016'
  AND date_of_payment::date <= '11/03/2017'
GROUP BY date_of_payment,
         applicable_manufacturer_or_applicable_gpo_making_payment_name,
         name_of_drug_or_biological_or_device_or_medical_supply_1,
         nature_of_payment_or_transfer_of_value;

GRANT ALL ON TABLE brams_payments TO redash;


SELECT *
FROM brams_payments;""", con=conn)

<IPython.core.display.Javascript object>

In [21]:
brams_payments["payments_amount"].sum()

184601.5

OK. It looks like Dr. Brams accepted nearly $185,000 from Alkermes and Otsuka to speak about their injectable anti-psychotic medications between January 2016 and November 3, 2017.

Next, how much drug company money did Arizona drug committee member Dr. Mohamed Ramadan accept?

In [22]:
%%notify
ramadan_payments = pd.read_sql("""
CREATE TABLE IF NOT EXISTS ramadan_payments AS
SELECT record_id,
       date_of_payment,
       nature_of_payment_or_transfer_of_value,
       name_of_drug_or_biological_or_device_or_medical_supply_1,
       sum(total_amount_of_payment_usdollars) AS payments_amount
FROM open_payments_general
WHERE physician_last_name = 'RAMADAN'
  AND physician_first_name = 'MOHAMED'
  AND recipient_state = 'AZ'
GROUP BY record_id,
         date_of_payment,
         nature_of_payment_or_transfer_of_value,
         name_of_drug_or_biological_or_device_or_medical_supply_1;

GRANT ALL ON TABLE ramadan_payments TO redash;


SELECT *
FROM ramadan_payments;""", con=conn)

<IPython.core.display.Javascript object>

In [23]:
ramadan_payments.groupby("nature_of_payment_or_transfer_of_value").agg({"record_id": len, "payments_amount": sum}).sort_values("payments_amount", ascending=False)

record_id  payments_amount
nature_of_payment_or_transfer_of_value                                        
Compensation for services other than consulting...        333        505174.21
Travel and Lodging                                        418        160918.49
Food and Beverage                                         447         23024.11
Consulting Fee                                             13         13063.69
Education                                                  15           485.11
Gift                                                        2           116.35

OK. It looks like Dr. Ramadan accepted nearly \$703,000 from drug companies, with more than \$505,000 coming in the form of compensation for giving speeches.

In [24]:
ramadan_payments.groupby("name_of_drug_or_biological_or_device_or_medical_supply_1").agg({"record_id": len, "payments_amount": sum}).sort_values("payments_amount", ascending=False)

record_id  payments_amount
name_of_drug_or_biological_or_device_or_medical...                            
ABILIFY MAINTENA                                          298        179779.54
Invega Sustenna                                           275        173574.11
LATUDA                                                    212        123602.91
INVEGA TRINZA                                             127         68074.70
INVEGA SUSTENNA                                           110         52920.15
Aristada 441 mg                                            72         50948.86
REXULTI                                                    37         25234.35
Latuda                                                     32         16772.19
SAPHRIS                                                     4          1989.25
BREXPIPRAZOLE                                              17          1055.54
ABILIFY TABLET                                              7           198.68
SUBOXONE SUBLINGUAL FILM                                    1           125.00
Brintellix                                                  2           100.46
No Product                                                  1           100.00
VRAYLAR                                                     3            98.32
Invega                                                      1            65.99
NUEDEXTA                                                    1            20.59
Vivitrol                                                    1            14.03
Trintellix                                                  1            10.13

And it looks like most of his payments related to two drugs, Abilify Maintena and Invega Sustenna. He received at least $406,293.80 in payments relating to the medications. Even though meeting minutes show Ramadan has missed nearly half of the Arizona drug committee meetings since he joined in 2016, he has shown up to the ones that reviewed Invega Sustenna and Abilify Maintena.

Finally, how much money related to his work as a consultant on Abilify Maintena did New York drug committee member Dr. James Saperstone accept prior to voting on the medication at a 2016 meeting?

In [25]:
%%notify
saperstone_payments = pd.read_sql("""
CREATE TABLE IF NOT EXISTS saperstone_payments AS
SELECT record_id,
       date_of_payment,
       nature_of_payment_or_transfer_of_value,
       name_of_drug_or_biological_or_device_or_medical_supply_1,
       sum(total_amount_of_payment_usdollars) AS payments_amount
FROM open_payments_general
WHERE physician_last_name = 'SAPERSTONE'
  AND physician_first_name = 'JAMES'
  AND recipient_state = 'NY'
GROUP BY record_id,
         date_of_payment,
         nature_of_payment_or_transfer_of_value,
         name_of_drug_or_biological_or_device_or_medical_supply_1;

GRANT ALL ON TABLE saperstone_payments TO redash;


SELECT *
FROM saperstone_payments;""", con=conn)

<IPython.core.display.Javascript object>

In [26]:
saperstone_payments[(saperstone_payments["name_of_drug_or_biological_or_device_or_medical_supply_1"].str.contains("abilify maintena", case=False, na=False)) & (saperstone_payments["nature_of_payment_or_transfer_of_value"] == "Consulting Fee")]

record_id date_of_payment nature_of_payment_or_transfer_of_value  \
7   105123594      10/20/2013                         Consulting Fee   
8   105123596      11/15/2013                         Consulting Fee   
35  216380760      05/19/2014                         Consulting Fee   
36  216380764      05/19/2014                         Consulting Fee   

   name_of_drug_or_biological_or_device_or_medical_supply_1  payments_amount  
7                                    ABILIFY MAINTENA                3600.00  
8                                    ABILIFY MAINTENA                 300.00  
35                                   ABILIFY MAINTENA                3300.00  
36                                   ABILIFY MAINTENA                  26.76

OK. It looks like Dr. Saperstone accepted $7,226.76 related to his work as a consultant on Abilify Maintena prior to voting on the medication in 2016.

## Export the data to Excel for visualization and further analysis

In [27]:
writer = pd.ExcelWriter("data/doc_payments.xlsx")
duplicates_removed.to_excel(writer, "Payments by Doctor", startcol=0, index=False)
payments_by_state.to_excel(writer, "Payments by State", startcol=0, index=False)
writer.save()